File: 2020 ETHCPC/Practice codes/basicprogramming1.py
n,t = [int(i) for i in raw_input().split()]
a = [int(i) for i in raw_input().split()]

if t == 1:
    print 7
elif t == 2:
    if a[0] > a[1]:print "Bigger"
    elif a[0] == a[1]:print "Equal"
    else: print "Smaller"
elif t == 3:
    r = a[:3]
    r.sort()
    print r[1]
elif t == 4:
    print sum(a)
elif t == 5:
    print sum([i for i in a if i%2 == 0] )
elif t == 6:
    s = ''
    
    for i in a:
        s += chr(97 + i%26)
    print s
elif t == 7:
    c = {i: False for i in a}
    i = 0
    c[0] = True
    while True:
        
        
        i = a[i]
        if i >= n:
            print 'Out'
            break
        if i == (n - 1):
            print 'Done'
            break
        if c[i]:
            print "Cyclic"
            break
        c[i] = True
        

File: Code site scrapping/kattis/archive/basicprogramming1.cpp
// Time: 2021-01-22 15:17:46
// title: Basic Programming 1
// language: C++


#include<bits/stdc++.h>
using namespace std;
int main(){
    int n,c,t;
    cin>>n>>t;
    int a[n];
    for(int i=0; i<n; i++)cin>>a[i];

    if(t == 1)cout<<7;
    else if(t == 2){
        if (a[0] > a[1])cout<<"Bigger";
        else if (a[0] == a[1]) cout<<"Equal";
        else cout<<"Smaller";

    }
    else if(t == 3){
        int r[] = {a[0], a[1], a[2]};
        sort(r, r+3);
        cout<< r[1];
    }
    else if (t == 4){
        long long s= 0;
        for(int i=0; i<n; i++)s += a[i];
        cout<<s;
    }
    else if (t == 5){
        long long s= 0;
        for(int i=0; i<n; i++)if(a[i] % 2 == 0)s += a[i];
        cout<<s;
    }
    else if (t == 6){
        string s = "";
        map<int, char>d;
        for(int i=0; i<26; i++) d[i] = char(97 + i);

        for(int i=0; i<n; i++)
            s += d[a[i]%26];
        cout<<s;
    }
    else if (t == 7){
        bool c[n];
        for(int i=0; i<n; i++) c[i] = false;
        int i = 0;
        while (true){
            i = a[i];
            if (i >= n){
                cout<<"Out";
                break;
            }
            else if (i == (n - 1)){
                cout<< "Done";
                break;
            }
            else if (c[i]){
                cout<< "Cyclic";
                break;
            }
            c[i] = true;
        }
    }
}
